File: Find_the_odd_int.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

int find_it(const std::vector<int>& seq) {
    std::unordered_map<int, int> element_count;
    
    // Преброяване на срещанията на всеки елемент в масива
    for (int element : seq) {
        if (element_count.find(element) != element_count.end()) {
            element_count[element] += 1;
        } else {
            element_count[element] = 1;
        }
    }
    
    // Намиране на елемента с нечетен брой срещания
    for (const auto& pair : element_count) {
        if (pair.second % 2 != 0) {
            return pair.first;
        }
    }
    
    return -1;
}

int main() {
    std::vector<int> seq = {1, 2, 3, 2, 1, 3, 1};
    int result = find_it(seq);
    std::cout << result << std::endl;
    
    return 0;
}

File: Find_the_odd_int.py
# Given an array of integers, find the one that appears an odd number of times.
# There will always be only one integer that appears an odd number of times.
def find_it(seq):
    element_count = {}
    
    # Преброяване на срещанията на всеки елемент в масива
    for element in seq:
        if element in element_count:
            element_count[element] += 1
        else:
            element_count[element] = 1
    
    # Намиране на елемента с нечетен брой срещания
    for element, count in element_count.items():
        if count % 2 != 0:
            return element
    
    return None

seq = [1, 2, 3, 2, 1, 3, 1]
result = find_it(seq)
print(result)  # Извежда: 2
